src/test/java/TasksServletTest.java

```diff
@@ -107,40 +108,49 @@ public void setUp() throws IOException, GeneralSecurityException {
 
   @Test
   public void noTaskLists() throws IOException, ServletException {
-    Mockito.when(tasksClient.listTaskLists()).thenReturn(NO_TASK_LISTS);
+    Mockito.when(tasksClient.listTaskLists()).thenReturn(NO_TASKLISTS);
     servlet.doGet(request, response);
     printWriter.flush();
-    Assert.assertTrue(stringWriter.toString().contains(EMPTY_JSON));
+    List<Task> tasks = gson.fromJson(stringWriter.toString(), LIST_OF_TASKS_TYPE);
+    Assert.assertTrue(tasks.isEmpty());
   }
 
   @Test
   public void emptyTaskLists() throws IOException, ServletException {
-    Mockito.when(tasksClient.listTaskLists()).thenReturn(SOME_TASK_LISTS);
-    Mockito.when(tasksClient.listTasks(Mockito.any())).thenReturn(NO_TASKS);
+    Mockito.when(tasksClient.listTaskLists()).thenReturn(SOME_TASKLISTS);
+    Mockito.when(tasksClient.listTasks(TASKLIST_ONE)).thenReturn(NO_TASKS);
+    Mockito.when(tasksClient.listTasks(TASKLIST_TWO)).thenReturn(NO_TASKS);
     servlet.doGet(request, response);
     printWriter.flush();
-    Assert.assertTrue(stringWriter.toString().contains(EMPTY_JSON));
+    List<Task> tasks = gson.fromJson(stringWriter.toString(), LIST_OF_TASKS_TYPE);
+    Assert.assertTrue(tasks.isEmpty());
   }
 
   @Test
   public void oneEmptyTaskList() throws IOException, ServletException {
-    Mockito.when(tasksClient.listTaskLists()).thenReturn(SOME_TASK_LISTS);
-    Mockito.when(tasksClient.listTasks(Mockito.any()))
-        .thenReturn(NO_TASKS)
-        .thenReturn(TASKS_ONE_TWO);
+    Mockito.when(tasksClient.listTaskLists()).thenReturn(SOME_TASKLISTS);
+    Mockito.when(tasksClient.listTasks(TASKLIST_ONE)).thenReturn(NO_TASKS);
+    Mockito.when(tasksClient.listTasks(TASKLIST_TWO)).thenReturn(TASKS_ONE_TWO);
     servlet.doGet(request, response);
     printWriter.flush();
-    Assert.assertTrue(stringWriter.toString().contains(TASK_ONE_TWO_JSON));
+    List<Task> tasks = gson.fromJson(stringWriter.toString(), LIST_OF_TASKS_TYPE);
+    tasks.forEach(
+        (task) -> {
+          Assert.assertTrue(TASKS_ONE_TWO.contains(task) || TASKS_THREE_FOUR.contains(task));
+        });
```

Comment:
Alternatively, you can sort both lists and compare with `Assert.assertEquals`.

Reply:
Alternatively, JUnit may have more advanced matchers to compare the elements of a list without order. https://github.com/junit-team/junit4/wiki/Matchers-and-assertthat

Reply:
Cool idea - I used the latter. Thanks!